ocaml/typing/typeopt.ml

```diff
@@ -579,18 +636,25 @@ let report_error ppf = function
   | Non_value_layout (ty, err) ->
       fprintf ppf
         "Non-value detected in [value_kind].@ Please report this error to \
-         the Jane Street compilers team.@ %a"
-        (Layout.Violation.report_with_offender
-           ~offender:(fun ppf -> Printtyp.type_expr ppf ty)) err
-  | Non_value_sort ty ->
+         the Jane Street compilers team.";
+      begin match err with
+      | None ->
+        fprintf ppf "@ Could not find cmi for: %a" Printtyp.type_expr ty
```

Comment:
This string does not appear in the testsuite. Maybe add a test to witness this behavior?

Reply:
I don't think this is reachable if the typechecker is correct.  `value_kind` would have to be called on a type that's not a value or for which we're missing the cmi directly (not in a recursive call).